pkg/cmd/server/origin/auth.go

```diff
@@ -379,6 +382,16 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 			if identityProvider.UseAsChallenger {
 				return nil, errors.New("oauth identity providers cannot issue challenges")
 			}
+		} else if requestHeaderProvider, isRequestHeader := identityProvider.Provider.Object.(*configapi.RequestHeaderIdentityProvider); isRequestHeader {
```

Comment:
fixed key collisions, added validation to prevent multiple redirectors or a mix of redirectors and challengers, added TODO to refactor this